1.5/Source/I_Aint_Building_That/Settings.cs

```diff
@@ -43,53 +51,101 @@ public void DoWindowContents(Rect wrect)
 			Listing_Standard options = new();
 			options.Begin(wrect);
 
-			if (options.ButtonText("Restore All Hidden Defs"))
+			switch (_page)
 			{
-				HiddenBuildables.Clear();
+				case Page.Buildings:
+					{
+						if (options.ButtonText("Restore All Hidden Defs"))
+						{
+							HiddenBuildables.Clear();
+						}
+
+						options.Gap();
+
+						// Add a TextField for the search query
+						string lastSearch = _searchQuery;
+						float searchBarY = wrect.y;
+						Widgets.Label(new Rect(0, searchBarY, 120, 30f), "Taggerung_IAintBuildingThat_SearchLabel".Translate());
+						_searchQuery = Widgets.TextField(new Rect(130, searchBarY, wrect.width - 130, 30f), _searchQuery);
+						if (lastSearch != _searchQuery) _scrollPosition = Vector2.zero;
+
+						options.Gap();
+
+						// Filter the current list
+						List<string> filteredBuildables = HiddenBuildables.Where(MatchesFilter).ToList();
+
+						// Create a scrollable list
+						Rect viewRect = new(0, 0, wrect.width - 16f, filteredBuildables.Count * RowHeight);
+						Rect scrollRect = new(0, 80f, wrect.width, wrect.height - 80f);
+						Widgets.BeginScrollView(scrollRect, ref _scrollPosition, viewRect);
+
+						float num = 0f;
+						foreach (string buildable in filteredBuildables)
+						{
+							BuildableDef def = LookupDef(buildable);
+							float rowSectionWidth = (viewRect.width - 20) / 2;
+							if (def != null)
+							{
+								Widgets.DefLabelWithIcon(new Rect(0f, num * RowHeight, rowSectionWidth, RowHeight - 5f), def);
+							}
+
+							Rect rowRect = def == null
+								? new Rect(0, num * RowHeight, viewRect.width, RowHeight - 5f)
+								: new Rect(rowSectionWidth + 10, num * RowHeight, rowSectionWidth, RowHeight - 5f);
+							if (Widgets.ButtonText(new Rect(rowRect.x + 10, rowRect.y, rowRect.width - 20f, rowRect.height), $"{"Taggerung_IAintBuildingThat_RestoreText".Translate()} {buildable}"))
+							{
+								HiddenBuildables.Remove(buildable);
+								break; // break to avoid collection modified exception
+							}
+
+							num++;
+						}
+
+						Widgets.EndScrollView();
+					} break ;
+				case Page.Abilities:
+					{
+						if (options.ButtonText("Restore All Hidden Defs"))
+						{
+							HiddenAbilities.Clear();
+						}
+
+						options.Gap();
+
+						// Create a scrollable list
+						Rect viewRect = new(0, 0, wrect.width - 16f, HiddenAbilities.Count * RowHeight);
+						Rect scrollRect = new(0, 80f, wrect.width, wrect.height - 80f);
+						Widgets.BeginScrollView(scrollRect, ref _scrollPosition, viewRect);
+
+						float num = 0f;
+						foreach (var ability in HiddenAbilities)
```

Comment:
A search bar would be helpful and I'd suggest sorting by Pawn, that way you can see all a pawn's skills together and filter on pawn name and ability label